File: Hot 100/300. 最长递增子序列.cpp
class Solution {
public:
    int lengthOfLIS(vector<int>& nums) {
        int n=nums.size();
        vector<int> dp(n ,0);
        if(n==0) {
            return 0;
        }
        for(int i=0; i<nums.size();i++){
            dp[i] = 1;
            for(int j=0;j<i;j++){
                if(nums[i] > nums[j]) {
                    dp[i] = max(dp[i], dp[j]+1);
                }
            }
        }
        return *max_element(dp.begin(),dp.end());
    }
};

File: 刷题/Hot 100/300. 最长递增子序列.go
func lengthOfLIS(nums []int) int {
    n := len(nums)
    dp := make([]int, n+1)
    for i:=0; i<n; i++ {
        dp[i] = 1
        for j:=0; j<i; j++ {
            if(nums[i] > nums[j]){
                dp[i] = max(dp[i], dp[j]+1)
            }
        }
    }
    ans := 0
    for i:=0; i<n; i++ {
        ans = max(ans, dp[i])
    }
    return ans
}

func max(a, b int) int {
    if a > b {
        return a
    }
    return b
}